### ISSUE TITLE
Semantic External Queries

### ISSUE THREAD

**Issue by JeroenDeDauw** [SemanticMediaWiki/SemanticExternalQueryLookup#20]
My suggestion for the name of this extension

**Comment by kghbln** [comment]
There is something to it. Dunno if this still can be done though

**Comment by mwjames** [comment]
I'm not sure the administrative work (renaming all stuff, repo, twn etc.) would be worth the effort.

**Comment by kghbln** [comment]
Yeah, second thoughts. Not really worth the trouble. If you like the idea behind the extension you will probably also memorise its name

**Comment by kghbln** [comment]
Once SMW 2.4 is out this could use a release too. :)

**Comment by JeroenDeDauw** [comment]
You could easily rename the repo (which leaves a redirect), the stuff in the README, and what is shown on Special:Version. If the code has a slightly different name, or the TWN thing is a bit different: #SuchShrugs. I figure people dealing with the code or TWN are unlikely to be bothered by this.

**Comment by kghbln** [comment]
I still like the name suggested but if a renaming or so blocks the release of this software I'd also take the current name. So no hold ups because of this ...

**Comment by mwjames** [comment]
> or so blocks the release of this software ...

For sometime I was unhappy about the requirement to inherit from the `SQLStore` as a means to build a contract to the `getQueryResult` method, now this precondition has ceased with the introduction of the `QueryEngine` interface by https://github.com/SemanticMediaWiki/SemanticMediaWiki/pull/1776.

**Comment by kghbln** [comment]
> now this precondition has ceased with the introduction of the QueryEngine interface by SemanticMediaWiki/SemanticMediaWiki#1776.

So I believe the first release will be done after the release of SMW 2.5 and will need it as the minimum requirement.
